File: mp/old/textparsing/flashcard_server.js
/**
 * Created by user1 on 2/7/14.
 *
 * serve up file, use flash cards to process it
 */
var fs = require("fs")
var shelpers = require('../proxy/helpers/shelpers').shelpers;
var request = require('request'),
    express = require('express');
var Flashcards = require('./flashcards.js').Flashcards;
var app = express();
app.use(express.json());       // to support JSON-encoded bodies
app.use(express.urlencoded());

var port = process.env.PORT || 9000;

app.configure(function () {
        app.set("view options", { layout: false, pretty: true });
        app.use(express.favicon());
        app.use("/public", express.static(__dirname + '/public'));
    }
);

app.listen(port);

// Routes
app.get('/', function (req, resp) {
    resp.render('index.jade', {pageTitle: 'Some title'});
});

app.get('/index.html', function (req, res) {
    res.sendfile('form.html')
})


app.post('/submit.html', function (req, res) {
    var formData = req.body.user;
    console.log(formData);
    var x = new Flashcards();
    x.initialize(null, formData);
    x.process();
    res.write(x.createTable());
    res.end();
})


app.get('/submit.html', function (req, res) {
    var formData = fs.readFileSync('oracle2.txt')
    console.log(formData)
    var x = new flashcards(formData)
    var results = x.process()
    for (var i = 0; i < results.length; i++) {
        var result = results[i];
        result = '<tr>' + '<td>' + result + '</td>' + '</tr>'
    }
    results = results.join("")
    results = results.replace("\t", '</td><td>')
    res.write('<html>' + '<table>' + results + '</table>' + '</html>')
    res.end()
})


app.get('/test', function (req, res) {
    var formData = fs.readFileSync('oracle2.txt')
    var x = new Flashcards()//, formData)
   // x.initialize('./oracle2.txt')
    x.initialize(null, formData)
    var results = x.process()
    res.write(x.createTable())
    res.end()
})

var open = require('open')
open('http://localhost:9000/index.html')
//http://localhost:9000/index.html

File: mp/old/textparsing/flashcard_server.rb


puts RUBY_VERSION
require 'rubygems'
require 'rack'
require 'sinatra'

require "sinatra/reloader"
STDOUT.sync = true

set :bind, 'localhost'
set :port, 7125

#puts RUBY_VERSION
#exit()
require_relative 'flashcards.rb'

get '/' do
	#raise File.dirname(__FILE__)
      filename = File.dirname(__FILE__)+'/'+'flashcards_html_form.html'
      displayfile = File.open(filename, 'r')
      content = displayfile.read()
      content
end

get '/test_require' do
	require_relative 'flashcards.rb'
	#include GoThroughFile
	puts "yo? you down? \n madrk?: fight".split('\n')
	go = GoThroughFile.new(nil, "yo? you down? \n madrk?: fight")
      go.process()
end

get '/evernote_eod_form.html' do
      filename = 'evernote_eod_form.html'
      displayfile = File.open(filename, 'r')
      content = displayfile.read()
      content
end

post '/submit.html' do
	require_relative 'flashcards.rb'
        input = params['user']
	input = input.gsub("\r", '' )        
	 puts 'input', input.inspect
	#puts "yo? you down? \n mark?: fight".split('\n')
	go = GoThroughFile.new(nil, input)
	results = go.process()#.join("\n")
	puts results.inspect
	results = results.collect do | x | 
		'<tr>'+'<td>'+x+'</td>'+'</tr>'
	end
	results = results.join("")
      #  result = result.gsub("\n", '<br />' )
      #  result = result.gsub("\r", '' ) 
	#result = result.gsub('\r', '' )  
	results = results.gsub("\t", '</td><td>' ) 	

	'<html>'+'<table>'+results+'</table>'+'</html>';      
end
